pkg/sync/sync.go

```diff
@@ -604,6 +647,13 @@ func worker(tasks <-chan object.Object, src, dst object.ObjectStorage, config *C
 	}
 }
 
+func checkUseServiceCopy(src, dst object.ObjectStorage) bool {
+	return reflect.TypeOf(src) == reflect.TypeOf(dst) &&
```

Comment:
Reflection is usually slow, and the check might be false positive if the object storages are wrapped by `withPrefix`.